baremetal/metal3data_manager.go

```diff
@@ -595,7 +595,7 @@ func (m *DataManager) addressFromM3Claim(ctx context.Context, poolRef corev1.Typ
 	}
 
 	if ipClaim.Status.ErrorMessage != nil {
-		m.Data.Status.ErrorMessage = pointer.StringPtr(fmt.Sprintf(
+		m.setError(ctx, fmt.Sprintf(
```

Comment:
Will it recover from this? Just checking since we know that if we bubble up an error all the way to the Machine, it will get stuck.

Reply:
m3m is  not failing when an error set here in m3d

Reply:
is there a reason we have to do it this way? in practice it is the same thing at the end: https://github.com/metal3-io/cluster-api-provider-metal3/blob/47af8777f5e218c0785b6d3e738522fd21188cca/baremetal/metal3data_manager.go#L105-L106

Reply:
The function was already declared so I just decided to use it at the end it is the same thing